PEP 622: Why the match statement is the correct feature right now (#1502)
python/peps: pep-0622.rst

with pattern matching, since although it may be handy, it will likely be used
rarely.

Other pattern-based constructions
---------------------------------

> Many other languages supporting pattern-matching use it as a base for multiple

Reviewer: ```suggestion
Many other languages supporting pattern-matching use it as a basis for multiple
```

language constructs, including a matching operator, a generalized form 
of assignment, a filter for loops, a method for synchronizing communication,
or specialized if statements. Some of these were mentioned in the discussion

> of the first draft. Another question asked is why this particular form (joining

Reviewer: ```suggestion
of the first draft. Another question asked was why this particular form (joining
```

binding and conditional selection) was chosen while other forms were not.

> Introducing more uses of patterns would be too bold and premature for the

Reviewer: ```suggestion
Introducing more uses of patterns would be too bold and premature given the
```

> experience we have using patterns and would make this proposal too

Reviewer: ```suggestion
experience we have using patterns, and would make this proposal too
```

complicated. The statement as presented provides a form of the feature that

> is sufficiently general to be useful while being self-contained and not

Reviewer: ```suggestion
is sufficiently general to be useful while being self-contained, and without
```

> having a massive impact in the syntax and semantics of the language as a whole.

Reviewer: ```suggestion
having a massive impact on the syntax and semantics of the language as a whole.
```

After some experience with this feature, the community may have a better

> knowledge of what other pattern matching constructs could be valuable in Python.

Reviewer: ```suggestion
feeling for what other uses of pattern matching could be valuable in Python.
```
